src/socket/tcp.rs

```diff
@@ -1396,6 +1406,9 @@ impl<'a> TcpSocket<'a> {
                 repr.control = TcpControl::Syn;
                 if self.state == State::SynSent {
                     repr.ack_number = None;
+                    repr.window_scale = Some(0);
+                } else {
+                    repr.window_scale = if self.remote_win_scale.is_some() { Some(0) } else { None }
```

Comment:
nit: not sure about this repo's coding style, but I personally prefer this: `repr.window_scale = self.remote_win_scale.map(|_| 0)`

Reply:
`.unwrap_or(0)`

Reply:
Fixed, @ProgVal.

Since this is an ``Option<u8>``, @whitequark, I'm not sure how ``unwrap_or()`` can create the same effect. Could you be more specific?